packages/jwt-middleware/index.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const Assert = require('assert');
+const ExpressJwt = require('express-jwt');
+const jsonwebtoken = require('jsonwebtoken');
+const JwksRsa = require('jwks-rsa');
+
+const WT_AUTH_EXEC = 'wt-authorize-execution';
+const WT_META_EXEC_SCOPE ='wt-execution-scope';
+const WT_META_ADMIN_SCOPE = 'wt:admin';
+const WT_META_ISS = 'wt-execution-iss';
+const WT_META_AUD ='wt-execution-aud';
+
+const AUTHZ = 'authorization';
+const BEARER = /^bearer\s+(.+)\s*$/i;
+
+module.exports = function middlewareFactory(){
+    let cachedMiddlewareInstance;
+
+    return function(req, res, next) {
+        if (!cachedMiddlewareInstance) {
+            cachedMiddlewareInstance = createMiddleware(req);
+        }
+
+        return cachedMiddlewareInstance(req, res, next);
+    }
+}
+
+function createMiddleware(req) {
+
+    const iss = req.webtaskContext.meta[WT_META_ISS];
```

Comment:
Do we need the request in order to get the context? I was under the impression that tomek said the metadata was already available within the middleware? @ggoodman --is that the case?